drivers/redis-driver/driver.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES LOSS OF USE, DATA, OR PROFITS OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict'
+
+var EventEmitter = require('events')
+
+
+/**
+ * options
+ * {
+ *   { host, port, list, redis - provide test redis client instance }
+ * }
+ * list should be service name, the driver will use two lists
+ * <list>_req
+ * <list>_res
+ * servers (services)  will listen on _req and send on _res
+ * clients (service invokers) will send on _req and listen on _res
+ */
+module.exports = function createRedisDriver (options) {
+  return function redisDriver (opts, cb) {
+    var emitter = new EventEmitter()
+    var tearingDown = false
+    var redis
+    var rin = null
+    var rout = null
+
+    if (opts instanceof Function) {
+      cb = opts
+    }
+    opts = opts || {}
+
+    emitter.on('receive', cb)
+
+    if (options.source && options.source.redis) {
```

Comment:
this assumes `options` will be an object, and it ultimately comes from user input, 
so if the user doesn't supply an options object, mu will blow up here 

perhaps in the case of a redis driver an options object will always be required, but we should give good error output, maybe `assert(options, 'redis driver requires an options object')`?